src/Data/Tree/AVL/Zipper.hs

```diff
@@ -150,20 +147,25 @@ runZipped action mode0 tree = do
 -- | Run zipper operation, build proof.
 runZipped' :: Retrieves h k v m => Zipped h k v m a -> Mode -> Map h k v -> m (a, Map h k v, Proof h k v)
 runZipped' action mode0 tree = do
-    zipper             <- enter mode0 tree
-    (a, tree1, trails) <- action' `evalStateT` zipper
-    proof              <- prune trails (fullRehash tree)
+    (a, tree1, trails) <- runZipped action mode0 tree
+    proof              <- prune trails tree
     return (a, tree1, proof)
-  where
-    action' = do
-      res    <- action
-      tree'  <- exit
-      trails <- use trail
-      return (res, tree', trails)
+
+-- say :: Retrieves h k v m => String -> Zipped h k v m ()
+-- say = liftIO . putStrLn
+
+-- dump :: Retrieves h k v m => Zipped h k v m ()
+-- dump = do
+--     say . showMap =<< use locus
+--     ctx <- use context
+--     say (show $ map (rootHash . _tzcFrom) ctx)
 
 -- | Materialise tree node at locus and give it to action for introspection.
 --
---   Often used with lambda-case: @withLocus $ \case { ... }@
+--   Often used with lambda-case:
+--
+--   > withLocus $ \case { ... }
+--
```

Comment:
Again, why not just delete that

Reply:
I won't delete these two functions, because each time I change something and something other breaks, I have to us'em to find what went wrong.

Reply:
I'm about `dump` and `say`.